### ISSUE TITLE
Session creation Failed - خطأ أثناء معالجة إتمام الطلب. يرجى المحاولة مرة أخرى

### ISSUE THREAD

**Issue by firasboubes** [PhazeRoOman/thawani-for-woocommerce#98]
### This issue addresses:
- Session creation Failed
- [ ] A suggestion for improvement

## Description:
I got the below message in log:

04-06-2023 @ 12:12:05 - 📝Attempting to create Session
04-06-2023 @ 12:12:06 - 📝Session creation Failed
04-06-2023 @ 12:12:06 - 📝Response Code: 
04-06-2023 @ 12:12:06 - 📝Response Description: 
04-06-2023 @ 12:12:06 - 📝Success URL: https://uatcheckout.thawani.om/pay/?key=DpnvEGdNmLSU0390KpfzV31lG8MgG4
04-06-2023 @ 12:12:06 - 📝Cancel URL: https://atiabalmalak.com/checkout/order-received/6528/?key=wc_order_RilcLMINPIWHj

#### What does this bug result in:
After selecting thawani payment from checkout page its failed and give  message "خطأ أثناء معالجة إتمام الطلب. يرجى المحاولة مرة أخرى"

#### How can this bug be recreated:
- A descriptive step by step guide as to how this issue can be recreated

#### Potential Fix (If cause known):
- If you reviewed the codebase any idea how this bug can be resolved

#### WordPress Environment Versions:
- WordPress Version: 6.2
- WooCommerce Version: 7.5.1
- PHP Version: 7.4.33

#### Description:
- What should this feature do. Be as descriptive as possible

#### Reason:
- How will this feature help our community
 What challenge does it tackle

**Comment by w7shdev** [comment]
Hey @firasboubes , I just tried to checkout with the development keys and it worked I don't see any issue.
also the plugin was able to generate session id which means there is an issue with your configuration in WP.

Here see this comment maybe it will help you to debug your issue https://github.com/PhazeRoOman/thawani-for-woocommerce/issues/90#issuecomment-1221318338 & make sure that the minimum product price is 100 Bisa.

**Comment by firasboubes** [comment]
Hi @w7shdev , thanks for your reply, yes you are right the development keys are working (with me also), but once I used the production keys for merchant accounts, the operation fails.
I am sure I did all the required steps as mentioned in the setup steps. also added Webhook URL.
From your experience do you have an idea what is the issue ?

**Comment by w7shdev** [comment]
> Hi @w7shdev , thanks for your reply, yes you are right the development keys are working (with me also), but once I used the production keys for merchant accounts, the operation fails. I am sure I did all the required steps as mentioned in the setup steps. also added Webhook URL. From your experience do you have an idea what is the issue ?

I see, the answer is in this comment https://github.com/PhazeRoOman/thawani-for-woocommerce/issues/90#issuecomment-1221318338 

**TLDR;**

You need to talk with Thawani pay team and activate the production account. _Thawani has to activate the production key, you can generate them but they are not activated_.

**Comment by w7shdev** [comment]
@firasboubes I'll close this issue for now.. if the issue you are facing is relating to the plugin please feel free to open this again..
